pkg/api/steve/catalog/content.go

```diff
@@ -108,10 +108,12 @@ func (i *contentDownload) serveIcon(apiContext *types.APIRequest, rw http.Respon
 	if err != nil {
 		return err
 	}
-	defer chart.Close()
+	if chart != nil {
+		_, err = io.Copy(rw, chart)
+		setIconHeaders(rw, suffix)
```

Comment:
You need to set the Icon headers first and then do the copying. Otherwise, the icons are not loading. please test.